inst/include/Rcpp/vector/no_init.h

```diff
@@ -35,9 +35,14 @@ class no_init_vector {
         return size;
     }
 
-    template <int RTYPE, template <class> class StoragePolicy >
-    operator Vector<RTYPE, StoragePolicy>() const {
-        return Rf_allocVector(RTYPE, size) ;
+    template <int RTYPE>
+    operator Vector<RTYPE, PreserveStorage>() const {
+        // Explicitly protect temporary vector to avoid false positive
+        // with rchk (#892)
+        SEXP x = PROTECT(Rf_allocVector(RTYPE, size));
```

Comment:
What about using `Shield<SEXP>` here rather than raw protect / unprotect?

Reply:
Need to check if `rchk` is satisfied.